waku/waku_rln_relay/rln_relay.nim

```diff
@@ -103,58 +103,54 @@ proc stop*(rlnPeer: WakuRLNRelay) {.async: (raises: [Exception]).} =
   await rlnPeer.groupManager.stop()
 
 proc hasDuplicate*(rlnPeer: WakuRLNRelay,
+                   epoch: Epoch,
                    proofMetadata: ProofMetadata): RlnRelayResult[bool] =
   ## returns true if there is another message in the  `nullifierLog` of the `rlnPeer` with the same
   ## epoch and nullifier as `proofMetadata`'s epoch and nullifier
   ## otherwise, returns false
   ## Returns an error if it cannot check for duplicates
 
-  let externalNullifier = proofMetadata.externalNullifier
   # check if the epoch exists
-  if not rlnPeer.nullifierLog.hasKey(externalNullifier):
+  let nullifier = proofMetadata.nullifier
```

Comment:
Maybe all this can be replace by something like (disregard types). `getOrDefault` might be useful.

```
return defaultTable.getOrDefault("epoch1", initTable[string, int]()).hasKey("key1")
```

A quick example would be something like this.
```
import tables

var defaultTable: Table[string, Table[string, int]] = {"epoch1":{"key1": 3, "key2": 3}.toTable,
                                                       "epoch2": {"key3": 3, "key4": 3}.toTable}.toTable

echo defaultTable.getOrDefault("epoch1", initTable[string, int]()).hasKey("key1")
# true
  
echo defaultTable.getOrDefault("epoch1", initTable[string, int]()).hasKey("key200")
# false

echo defaultTable.getOrDefault("epoch5", initTable[string, int]()).hasKey("key1")
# false
```

just an idea though, at your discretion.

Reply:
addressed in 6d20aef7

Reply:
not sure thats addressed there. in any case not blocking.

Reply:
we don't actually need to get the value from the nullifierLog table, just check if it exists.